core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/UnitOfWorkTransactionContext.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.user.core.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import javax.sql.DataSource;
+
+public class UnitOfWorkTransactionContext {
+
+    private static Log log = LogFactory.getLog(UnitOfWorkTransactionContext.class);
+    private HashMap<DataSource, Connection> activeConnection = new HashMap();
+    private int transactionDepth = 0;
+
+    /**
+     * Commit all the transaction, if something happened at the middle then it print all the commit and uncommitted
+     * transaction.
+     */
+    public void commitAllConnection() {
+
+        List<DataSource> listOfDataSource = new ArrayList<>();
+        for (DataSource dataSource : activeConnection.keySet()) {
+            try {
+                Connection connection = activeConnection.get(dataSource);
+                if (connection != null) {
+                    listOfDataSource.add(dataSource);
+                    connection.commit();
+                }
+            } catch (SQLException e) {
+
+                log.error("Error occurred while committing the connection. Connection: " + activeConnection
```

Comment:
We may not need listOfDataSource as we log each error inside the loop

Reply:
Ideally we need to throw exception when we can not commit all, so that caller can make a decision.
The exception should have list of datasources which failed the commit.